### ISSUE TITLE
Automate Git tag creation for Android apps

### ISSUE THREAD

**Issue by nya-elimu** [elimu-ai/wiki#130]
For each Android app, add a GitHub Action workflow for automatically performing a Git tag creation each time a pull request is merged into `main`, essentially performing these steps:

1. Remove `-SNAPSHOT` from `versionName`
2. Commit changes
3. Create Git tag
4. Bump `versionCode` and `versionName`, and add `-SNAPSHOT` to `versionName`
5. Commit changes

**Comment by vivekkiran** [comment]
I am interested in this task.

**Comment by nya-elimu** [comment]
@vivekkiran Great! Sent you an invite to join the GitHub organization 🙂

**Comment by nya-elimu** [comment]
@vivekkiran Are you still interested?

**Comment by nya-elimu** [comment]
@vivekkiran? 🙂

**Comment by tuancoltech** [comment]
@jo-elimu Considering that this plugin https://github.com/ajoberstar/grgit is not maintained any longer, and will soon be oudated, should we use Github Packages for automatic release instead?
If yes, I can help configure that.

**Comment by jo-elimu** [comment]
@tuancoltech Yes, please research if that is something we can do for all of our Android apps. Thank you.

Also be aware that we are using JitPack for release management of the `model` and `webapp` repos: http://jitpack.io/#elimu-ai.

**Comment by jo-elimu** [comment]
@tuancoltech Yes, but does that release script also work for the `utils` module as well?

**Comment by jo-elimu** [comment]
@tuancoltech From what I remember, we just edited the version in the `build.gradle` files manually for every release. We tried running the script for automating it, but we couldn't get it to work fully.

**Comment by tuancoltech** [comment]
> [@tuancoltech](https://github.com/tuancoltech) From what I remember, we just edited the version in the `build.gradle` files manually for every release. We tried running the script for automating it, but we couldn't get it to work fully.

@jo-elimu Then, I think we should first implement similar bumping script for `utils` module first, before thinking of automatic tasks for both modules.
@coderabbitai: Could you create a task for implementing version bumping via Gradle task for `utils` module in this repo: https://github.com/elimu-ai/content-provider and assign to me?

**Comment by jo-elimu** [comment]
> Then, I think we should first implement similar bumping script for `utils` module first, before thinking of automatic tasks for both modules.

@tuancoltech Yes, that makes sense. Let's make it work in one app module first. And after we got it working well, we can replicate the setup in other Android apps.
